geps/gep-713/index.md

```diff
@@ -3,1155 +3,1114 @@
+* **Scalar values (“Patch”):** the specs of 2 metaresources are merged into one by applying the winning spec (according to semantics dictated by the base merge strategy, i.e., the more specific if Defaults or the less specific one if Overrides) over the other spec, in a JSON patch operation.
+* **\<Custom>:** the spec of 2 metaresources are mixed into a composition of both specs, following a custom merge algorithm specified by the metaresource or policy kind.
 
-### The Problem, restated
-What this parable makes clear is that, in the absence of information about what
-Policy is affecting an object, it’s very easy to make poor decisions.
+The final set of *merge strategies* therefore supported by a metaresource CRD (base merge strategy \+ atomicity level) is any subset of the following, where \<Custom> is implementation-specific:
+* None
+* Atomic Defaults
+* Atomic Overrides
+* Patch Defaults
+* Patch Overrides
+* \<Custom> Defaults
+* \<Custom> Overrides
 
-It’s critical that this proposal solve the problem of showing up to three things,
-listed in increasing order of desirability:
+Metaresource kinds MAY opt to implement any of these strategies, including multiple strategies.
 
-- _That_ some Policy is affecting a particular object
-- _Which_ Policy is (or Policies are) affecting a particular object
-- _What_ settings in the Policy are affecting the object.
+Metaresource kinds that do not specify any merge strategy and only support targeting a single kind (with Declared target = Effective target), by default MUST implement the **None** merge strategy. (See the definition of [Direct](#direct) class of metaresources.)
 
-In the parable, if Ana and Chihiro had known that there were Policies affecting
-the relevant object, then they could have gone looking for the relevant Policies
-and things would have played out differently. If they knew which Policies, they
-would need to look less hard, and if they knew what the settings being applied
-were, then the parable would have been able to be very short indeed.
+Metaresource kinds that do not specify any merge strategy and support targeting multiple effective kinds, by default MUST implement the **Atomic Defaults** merge strategy.
 
-(There’s also another use case to consider, in that Chihiro should have been able
-to see that the Policy on the namespace was in use in many places before deleting
-it.)
+Metaresource kinds that implement more than one merge strategy MUST define a clear structure for the instances of metaresource to specify which of the supported strategies to apply. Instances of these metaresources MUST NOT be allowed to declare more than one merge strategy at a time, but only one of the supported strategies. If no merge strategy is specified by a given instance of the metaresource, the **Atomic Defaults** merge strategy MUST be assumed.
```

Comment:
> Instances of these metaresources MUST NOT be allowed to declare more than one merge strategy at a time, but only one of the supported strategies

What does this mean? Are you referring to 2 policy CRs having different merge strategies?

Reply:
I meant one CR can only declare one merge strategy. E.g., if the CRD was designed to enable specifying `defaults` and `overrides` merge strategies, it's either `defaults` or `overrides` in any given CR, but not both simultaneously.

Two CRs of the same CRD can declare different merge strategies between them, if the CRD supports multiple specifying multiple strategies. E.g., one CR specifies `defaults`, another CR specifies `overrides`.

I'll rephrase it to avoid any confusion.

Reply:
There is no hard specification for a `mergeStrategy` field, so to speak. Implementations are free to come up with their own custom strategy names, field names to specify the chosen strategy in a CR, and even schema (should a single field not be enough to specify it). IOW, it is an enum.